impeller/tessellator/tessellator.cc

```diff
@@ -78,51 +78,132 @@ Tessellator::Result Tessellator::Tessellate(
   constexpr int kVertexSize = 2;
   constexpr int kPolygonSize = 3;
 
-  //----------------------------------------------------------------------------
-  /// Feed contour information to the tessellator.
-  ///
-  static_assert(sizeof(Point) == 2 * sizeof(float));
-  for (size_t contour_i = 0; contour_i < polyline.contours.size();
-       contour_i++) {
-    size_t start_point_index, end_point_index;
-    std::tie(start_point_index, end_point_index) =
-        polyline.GetContourPointBounds(contour_i);
-
-    ::tessAddContour(tessellator,  // the C tessellator
-                     kVertexSize,  //
-                     polyline.points.data() + start_point_index,  //
-                     sizeof(Point),                               //
-                     end_point_index - start_point_index          //
+  if (polyline.contours.size() > 30 && fill_type == FillType::kNonZero) {
```

Comment:
Consider pulling 30 out to a named constant, and as much as might be possible, explain why 30 was chosen.

Reply:
I'd also add a comment about not using indices since there is a lot of code to read through to understand that's the case.

Reply:
Added more comments here.